x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/filters/filter_popover.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { MouseEventHandler, useState, useRef } from 'react';
+import { EuiPopover, EuiFieldText, EuiForm, EuiFormRow, keys } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { Filter, emptyFilter, SEARCH_QUERY_LANGUAGE } from '.';
+import { IndexPattern } from '../../../types';
+
+import {
+  QueryStringInput,
+  Query,
+  esKuery,
+  esQuery,
+} from '../../../../../../../../src/plugins/data/public';
+
+const defaultPlaceholderMessage = i18n.translate(
+  'xpack.lens.indexPattern.filters.label.placeholder',
+  {
+    defaultMessage: 'All requests',
+  }
+);
+
+export const FilterPopover = ({
+  filter,
+  setFilter,
+  indexPattern,
+  Button,
+}: {
+  filter: Filter;
+  setFilter: Function;
+  indexPattern: IndexPattern;
+  Button: React.FunctionComponent<{ onClick: MouseEventHandler }>;
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [tempFilter, setTempFilter] = useState(filter);
+
+  const inputRef = useRef<HTMLInputElement>();
+  const onQueryChange = (input: Query) => setTempFilter({ ...tempFilter, input });
+  const onLabelChange = (label: string) => setTempFilter({ ...tempFilter, label: label.trim() });
+  const onSubmit = (input: Query) => {
+    try {
+      if (input.language === SEARCH_QUERY_LANGUAGE.KUERY) {
+        esKuery.toElasticsearchQuery(esKuery.fromKueryExpression(input.query), indexPattern);
+      } else if (input.language === SEARCH_QUERY_LANGUAGE.LUCENE) {
+        esQuery.luceneStringToDsl(input.query);
+      }
+      if (input.query.length) {
+        if (tempFilter.label) {
+          setFilter(tempFilter);
+          setIsPopoverOpen(false);
+        } else {
+          if (inputRef.current) inputRef.current.focus();
+        }
+      }
+    } catch (e) {
+      console.log('Invalid syntax', JSON.stringify(e, null, 2)); // eslint-disable-line no-console
+    }
+  };
+
+  return (
+    <EuiPopover
+      anchorClassName="lnsLayerPanel__anchor"
+      panelClassName="lnsIndexPatternDimensionEditor__filtersEditor"
+      ownFocus
+      isOpen={isPopoverOpen}
+      closePopover={() => {
+        setIsPopoverOpen(false);
+        setTempFilter(filter);
```

Comment:
The behavior of this popover isn't consistent with the rest of Lens where autosaving is expected. I would expect that closing the popover actually has no effect, because anything edited in the popover would have immediately saved.